### PR TITLE
Port from Python2 to Python3.

### PR DESCRIPTION
**Pull request by meyersbs**
### Python:
* General syntax updates for changes from Python2 to Python3
* Code comments have been significantly updated
* `cPickle` --> `pickle`

### Dependencies:
* `scipy==0.12.0` --> `scipy==0.18.1`
* `scikit-learn==0.15.1` --> `scikit-learn==0.18.1`
* `numpy==1.9.0` --> `numpy==1.12.0`
* `nltk==3.0.0` --> `nltk==3.2.1`

### Support Data:
* There was a since nonstandard unicode character (the accented <b><em>i</em></b> in <b><em>naive</em></b>) in the list of negative words. I replaced this character with the correct escape sequence. To properly parse the escape sequence, `codecs.open` was used instead of the standard `open` for this file.

### Classification Model:
* The use updated `pickle` and `scikit-learn` libraries caused quite a hassle:
    * Classification models created using `scikit-learn` prior to version `0.18.1`, did not embed the `scikit-learn` version into model. Since `0.18.1` does embed its version number, it gave a number of warnings about loading an <b><em>outdated</em></b> model.
    * I tried countless methods, but Python3's `pickle` simply cannot derserialize an object that was serialized using Python2's `cPickle`.
* In response to these conjoined errors, I decided to retrain the classification model using the original dataset. Since this required generating dependency parses for each request in the dataset, I chose to dump the annotated dataset, <em>with dependency parses</em>, as a json dictionary in the format expected by `/scripts/train_model.py`. See the `README` for information about how to download these datasets (they are too large for version control).
* The retrained model is included in this pull-request. Since the json formatted training datasets I have provided are so large, retraining the model will require a significant amount of memory.

### Original Datasets:
* I have included the original datasets in the `corpora/` directory.

### Helper Functions:
* I have created an additional file, `corpora/download.py`,  which automates downloading and extracting the json formatted training data.

### CODE REVIEW

**Comment by cristiandnm** [comment]
Hi Benjamin,

Thanks so much for working on this!  This is a much-needed clean-up.  For legacy reasons we need to make sure that the code we use in the original paper is still available, and that the new model produces similar (or better) classification accuracy than the old one.  One option would be to keep both in the repository.  An old legacy python2 version and a new python3 version.  

What do you think?

**Comment by meyersbs** [comment]
Thank you for getting back to me! I completely understand the need to keep the legacy code alive, so here are the options that I see:

1) We tag this repository in it's current state as a 'Python2 Release'. Then we merge my pull-request and tag the repository as a 'Python3 Release'. I think this would be fine, since I doubt there will be any further development for Python2?
2) If you do plan further development in Python2, I think our best bet would be to create a new branch in this repository. Master can be Python2 and the new branch can be Python3. Then we can just denote in the README that there is a Python3 branch.
3) We could keep both versions in the same repository, but I foresee that getting messy quickly.
4) Finally, we could just keep our repositories separate and link to each other in our respective READMEs.

I am open to other suggestions. Let me know what you think!

- Ben

**Comment by cristiandnm** [comment]
Hi Ben, thanks for the suggestions.  Let's play it safe and go with option 2, and change the readme to have the note about the new version/branch very salient (at the top), with the appropriate acknowledgement.

**Comment by meyersbs** [comment]
That sounds great! Let me know when you've created the new branch and I'll resubmit my pull-request to the new branch.

**Comment by cristiandnm** [comment]
done, the branch is named python3

**Comment by meyersbs** [comment]
I have submitted the new pull-request.

**Comment by cristiandnm** [comment]
(I updated the readme and emailed you at bsm9339@rit.edu)
